src/CalcViewModel/Common/NavCategory.cpp

```diff
@@ -49,151 +44,258 @@ static constexpr int CURRENCY_ID = 16;
 static constexpr int GRAPHING_ID = 17;
 // ^^^ THESE CONSTANTS SHOULD NEVER CHANGE ^^^
 
+wchar_t* towchar_t(int number)
+{
+    auto wstr = to_wstring(number);
+    return _wcsdup(wstr.c_str());
+}
+
+extern "C"
+{
+    WINADVAPI LSTATUS APIENTRY RegOpenKeyW(_In_ HKEY hKey, _In_opt_ LPCWSTR lpSubKey, _Out_ PHKEY phkResult);
+#define RegOpenKey RegOpenKeyW
+    WINADVAPI LSTATUS APIENTRY RegQueryValueExW(
+        _In_ HKEY hKey,
+        _In_opt_ LPCWSTR lpValueName,
+        _Reserved_ LPDWORD lpReserved,
+        _Out_opt_ LPDWORD lpType,
+        _Out_writes_bytes_to_opt_(*lpcbData, *lpcbData) __out_data_source(REGISTRY) LPBYTE lpData,
+        _When_(lpData == NULL, _Out_opt_) _When_(lpData != NULL, _Inout_opt_) LPDWORD lpcbData);
+    WINADVAPI LSTATUS APIENTRY RegCloseKey(_In_ HKEY hKey);
+#define RegQueryValueEx RegQueryValueExW
+}
+
+bool IsGraphingModeAvailable()
+{
+    static bool supportGraph = Windows::Foundation::Metadata::ApiInformation::IsMethodPresent("Windows.UI.Text.RichEditTextDocument", "GetMath");
+    return supportGraph;
+}
+
+Box<bool> ^ _isGraphingModeEnabledCached = nullptr;
+bool IsGraphingModeEnabled()
+{
+    if (!IsGraphingModeAvailable())
+    {
+        return false;
+    }
+
+    if (_isGraphingModeEnabledCached != nullptr)
+    {
+        return _isGraphingModeEnabledCached->Value;
+    }
+    HKEY key;
+    DWORD allowGraphingCalculator{ 0 };
+    DWORD bufferSize{ sizeof(allowGraphingCalculator) };
+    // Make sure to call RegOpenKey only on Windows 10 1903+
+    if (RegOpenKey(HKEY_LOCAL_MACHINE, TEXT("SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\Calculator"), &key) != ERROR_SUCCESS)
+    {
+        _isGraphingModeEnabledCached = true;
+        return true;
+    }
+    if (RegQueryValueEx(key, TEXT("AllowGraphingCalculator"), NULL, NULL, reinterpret_cast<LPBYTE>(&allowGraphingCalculator), &bufferSize) != ERROR_SUCCESS)
```

Comment:
Consider using RegGetValueW instead:

RegGetValueW(HKEY_LOCAL_MACHINE, L"Software\\...\\\Calculator", L"AllowGraphingCalculator", RRF_RT_DWORD, ...)

That way you can do this in a single call and avoid dealing with handle lifetime management. Also you can enforce that the key is of the expected type.

Reply:
Also, don't use the TEXT() macro in code that doesn't build for both Unicode and ASCII. Just use L""